design-documents/graph-ql/coverage/customer/gift-registry.graphqls

```diff
@@ -128,40 +130,63 @@ input GiftRegistryDynamicAttributeInput {
     value: String!
 }
 
-type CreateGiftRegistryOutput {
-    gift_registry: GiftRegistry
+interface GiftRegistryUserErrorsInterface {
+    status: Boolean!
+    user_errors: [GiftRegistryItemsUserError]!
+}
+
+interface GiftRegistryItemUserErrorInterface {
+    status: Boolean!
+    user_errors: GiftRegistryItemsUserError
 }
 
-type UpdateGiftRegistryOutput {
+interface GiftRegistryOutputInterface {
     gift_registry: GiftRegistry
 }
 
+type CreateGiftRegistryOutput implements GiftRegistryOutputInterface {
+}
+
+type UpdateGiftRegistryOutput implements GiftRegistryOutputInterface  {
+}
+
 type RemoveGiftRegistryOutput {
     is_removed: Boolean!
 }
 
-type AddGiftRegistryItemsOutput {
-    gift_registry: GiftRegistry
+type AddGiftRegistryItemsOutput implements GiftRegistryOutputInterface, GiftRegistryItemUserErrorInterface {
 }
 
-type RemoveGiftRegistryItemsOutput {
-    gift_registry: GiftRegistry
+type MoveCartItemsToGiftRegistryOutput implements GiftRegistryOutputInterface, GiftRegistryUserErrorsInterface {
 }
 
-type UpdateGiftRegistryItemsOutput {
-    gift_registry: GiftRegistry
+type GiftRegistryItemsUserError {
+    message: String!
+    product_uid: ID
+    gift_registry_item_uid: ID
```

Comment:
gift_registry_item_uid and gift_registry_uid are optional - in case of MoveCartItemsToGiftRegistry we might not have either
product_uid is also optional because we might refer to a product that doesn't exist anymore

In case we have a valid reference these should be populated

Reply:
Will they never be populated on `MoveCartItemsToGiftRegistry` at all?